### PR TITLE
Fatal Realtime Service Error

### PR DESCRIPTION
**Pull request by kyscott18**

### CODE REVIEW

File: packages/core/src/sync-realtime/service.ts [hunk 1]
```diff
@@ -251,15 +252,16 @@ export class RealtimeSyncService extends Emittery<RealtimeSyncEvents> {
...
-        msg: `Realtime sync task failed (network=${
-          this.network.name
-        }, error=${`${error.name}: ${error.message}`})`,
+        error,
+        msg: `Realtime sync task failed (network=${this.network.name})`,
         network: this.network.name,
       });
-      // Note: Good spot to throw a fatal error.
+
+      this.emit("fatal", error);
```
Comment (0xOlias) [inline packages/core/src/sync-realtime/service.ts:264]: Are there any retries before this happens?

File: packages/core/src/Ponder.ts [hunk 2]
```diff
@@ -549,6 +549,16 @@ export class Ponder {
       realtime.on("shallowReorg", (checkpoint) => {
         this.syncGatewayService.handleReorg(checkpoint);
       });
+
+      realtime.on("fatal", async (error) => {
+        error.stack = undefined;
+        this.common.logger.fatal({
+          service: "app",
+          msg: "Failed to fetch initial realtime data",
```
Comment (0xOlias) [inline packages/core/src/Ponder.ts:557]: Fix msg

File: packages/core/src/sync-realtime/service.ts (hunk 1 above)
Comment (kyscott18) [reply packages/core/src/sync-realtime/service.ts:264]: Right now just the internal viem transport retries

File: packages/core/src/sync-realtime/service.ts [hunk 3]
```diff
@@ -245,31 +246,35 @@ export class RealtimeSyncService extends Emittery<RealtimeSyncEvents> {
...
+          }, error=${`${error.name}: ${error.message}`})`,
+          network: this.network.name,
+        });
 
-      if (this.isProcessBlockQueued) {
-        this.isProcessBlockQueued = false;
-        await this.process();
-      } else {
-        this.emit("idle");
+        if (i === 3) this.emit("fatal");
```
Comment (0xOlias) [inline packages/core/src/sync-realtime/service.ts:266]: Looks great

**Review by 0xOlias — APPROVED** [review]

**Pull request merged**